packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/AxesOptions.tsx

```diff
@@ -142,6 +158,19 @@ const AxesOptions: FC<Props> = ({ itemsMap }) => {
                 }
                 onBlur={(e) => setXAxisName(e.currentTarget.value)}
             />
+            {isXFieldDateOrNumber() && (
```

Comment:
Since we run this function on each render, we could use a `useMemo` instead.

Reply:
yeah, good idea, however, this function depends on a few variables that are created after 
>     if (!isCartesianVisualizationConfig(visualizationConfig)) return null;
and we can't `error  React Hook "useMemo" is called conditionally.`

Which means we'll have to move everything before `if (!isCartesianVisualizationConfig(visualizationConfig)) return null;` which may be even less performant

What do you think ?